streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -752,6 +762,7 @@ long runOnce(final long recordsProcessedBeforeCommit) {
 
         if (state == State.PARTITIONS_ASSIGNED) {
             if (taskManager.updateNewAndRestoringTasks()) {
+                updateThreadMetadata(taskManager.activeTasks(), taskManager.standbyTasks());
```

Comment:
This seems to conflict with my comment from above. If we set new tasks for `PARTITION_ASSIGNED` here, why not above?

Reply:
As the updateThreadMetadata() is directly followed by the setState() method this call should be removed. In that way, updateThreadMetadata can be invoked by setState(). Thank you for your comments